typings/react-dropzone.d.ts

```diff
@@ -39,7 +39,7 @@ export type DropzoneOptions = Pick<React.HTMLProps<HTMLElement>, PropTypes> & {
   noDrag?: boolean;
   noDragEventsBubbling?: boolean;
   disabled?: boolean;
-  onDrop?: <T extends File>(
+  onDrop?: <T extends FileWithPath>(
```

Comment:
`T` can be `FileWithPath` (you could either typecast or use `FileWithPath` as an arg to the cb function and typescript should pick up on it). It was intentional to keep the lower level type here instead of the `FileWithPath` because the type depends on what users use for the `getFilesFromEvent` func.